### PR TITLE
Fix Utils.Misc

### PR DESCRIPTION
**Pull request by iacore**
I'm not sure what the code does, but I made it work with latest Idris2.

### CODE REVIEW

File: src/Utils/Misc.idr [hunk 1]
```diff
@@ -96,7 +96,7 @@ export
 split_at_concat_rev : (n : Nat) -> (xs : Vect (n + m) a) -> {0 l : Vect n a} -> {0 r : Vect m a} -> (0 _ : splitAt n xs = (l, r)) -> l ++ r = xs
 split_at_concat_rev Z _ Refl = Refl
 split_at_concat_rev (S n) (x::xs) {l} prf with (splitAt n xs) proof sprf
-  split_at_concat_rev (S n) (x::xs) {l=x::l} Refl | (l, r) = cong (x::) $ split_at_concat_rev n xs sprf
```
Comment (iacore) [inline src/Utils/Misc.idr]: `r` is bound to `Vect` of multiplicity of 0. Here it should be type of vector element.

Comment (DSLstandard) [reply src/Utils/Misc.idr]: I don't quite get what you mean. I'm sorry... but if I'm not mistaken, I think `(l, r)` corresponds to the result of `splitAt n xs`, which returns the left splice, and the right splice of `xs` when splitted at index `n`. Also, changing `r` to `element` will result in:
```
Pattern variable element unifies with: ?r [no locals in scope].
```

**Comment by DSLstandard** [comment]
ive just updated to the latest commit at the moment - 43c5b5263cf7dd6788291b73220d841660050407 (Idris 2, version 0.5.1-47cae3459), and it works fine for me if i didn't make a mistake.

Q: what the code does?
A: im going to add the comment "to prove that, say given `xs : Vect (n + m) a`, can be splitted at index `n` into `l` and `r` (the (0 _ : splitAt n xs = (l, r)) part), and then be concatenated back into xs"

**Comment by ghost** [comment]
This is very strange. Both me and @tensorknower69 are able to compile on our local machines, but the error you described is showing up on CI.

**Comment by ghost** [comment]
In the end this is resolved by changing `element` to `_`. This is a very strange problem since both me and @tensorknower69 couldn't reproduce the problem, and compiling your branch on our machines actually caused a compilation error. However, GitHub CI was able to reproduce the error and your PR fixed the problem on Github CI. 

It's almost as if we are all running different versions of Idris. I don't really understand why this is the case, but using `_` fixed the problem, so I guess it worked out in the end.

**Comment by DSLstandard** [comment]
Apparently I screwed up massive time, I didn't even notice that `43c5b5263cf7dd6788291b73220d841660050407 (Idris 2, version 0.5.1-47cae3459)` makes no sense. Now that I have truly updated my Idris2, the original implementation fails, changing to `_` works.

**Comment by iacore** [comment]
I'm late, and you figured it out! I think the newset Idris 2 version also binds implicit arguments inside `{0 r : Vect m a}`.

**Pull request merged**